### PR TITLE
Add marquee view for running time entry description

### LINKED ISSUE #291: Animations: Running entry description
- When running entry's description/task/project/client doesn't fit into the area shown below, animate it to show the whole description/task/project/client.

screen shot 2017-07-19 at 10 53 56

toggl-ios-running-entry

```javascript
moveText = () ->
	toggl.text.animate
		x: newX
		options:
			time: toggl.text.width/60
			curve: Linear

Linear = "cubic-bezier(0, 0, 1, 1)"
```

### PR DESCRIPTION
**Pull request by arturs105**
Closes #291 

Please do a functional test with long and short descriptions.

### CODE REVIEW

File: Toggl.Daneel/Views/MarqueeView.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            else
+                startAnimation();
+        }
+
+        private void startAnimation()
+        {
+            var contentWidth = ContentSize.Width;
+            var end = new CGPoint(contentWidth - Bounds.Width, 0);
+
+            var duration = contentWidth / 60;
```
Comment (simonrozsival) [inline Toggl.Daneel/Views/MarqueeView.cs]: Please use a constant instead of a magic literal.

File: Toggl.Daneel/Views/MarqueeView.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            var durationWithPause = duration + 2;
```
Comment (simonrozsival) [inline Toggl.Daneel/Views/MarqueeView.cs]: Also a constant.

File: Toggl.Daneel/Views/MarqueeView.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+            var relativeStart = 1 / durationWithPause;
+            var relativeLength = 1 - 2 * relativeStart;
+
+            AnimateKeyframes(
+                duration: durationWithPause,
+                delay: 1,
```
Comment (simonrozsival) [inline Toggl.Daneel/Views/MarqueeView.cs]: Make a constant.

File: Toggl.Daneel/Views/MarqueeView.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                //3 << 16 because using UIViewKeyframeAnimationOptions.CalculationModeLinear doesn't work.
+                //Thanks Apple
+                options: UIViewKeyframeAnimationOptions.Repeat
+                    | UIViewKeyframeAnimationOptions.Autoreverse
+                    | (UIViewKeyframeAnimationOptions)(3 << 16),
```
Comment (simonrozsival) [inline Toggl.Daneel/Views/MarqueeView.cs]: Make your own constant then.

File: Toggl.Daneel/Views/MarqueeView.cs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                animations: () =>
+                {
+                    AddKeyframeWithRelativeStartTime(relativeStart, relativeLength, () => ContentOffset = end);
+
```
Comment (simonrozsival) [inline Toggl.Daneel/Views/MarqueeView.cs]: Empty line?

File: Toggl.Daneel/ViewControllers/MainViewController.cs [hunk 6]
```diff
@@ -164,5 +168,40 @@ private void animateSpider()
...
         }
+
+        private void addFading()
+        {
+            var gradient = new CAGradientLayer();
+            var fadingWidth = 8;
+            var relativeFadingStart
+                = fadingWidth / RunningEntryDescriptionFadingView.Bounds.Width;
+
+
```
Comment (simonrozsival) [inline Toggl.Daneel/ViewControllers/MainViewController.cs]: Extra empty line.

**Comment by simonrozsival** [comment]
The marquee effect works and looks nice 👍 So please introduce those constants and we can merge this!

**Comment by amulware** [comment]
Functional test shows:

1. Even if the text is in the right-most position, it still fades the beginning of it

screen shot 2017-10-20 at 09 55 14

2. After changing the description the animation does not update and looks broken

screen shot 2017-10-20 at 09 56 03

It probably should update/reset whenever the description changes. I think a simple reset is fine, not sure we need something more complicated.

**Pull request merged**